<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>CV</title>

<script src="site_libs/header-attrs-2.4.0/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery-3.5.1.min.js"></script>
<meta name="name" content="viewport" />
<meta name="content" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
<link href="site_libs/bootstrap-3.4.1/bootstrap-custom.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.4.1/bootstrap.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre {
    background-color: #f5f5f5;
    border: 1px solid #ccc;
    border-radius: 4px;
    padding: .5rem;
  }
  pre:not([class]) {
    background-color: white;
  }
  pre code {
    background-color: transparent;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}

.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* TODO: remove this padding CSS when we drop BS3 */
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}

.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!--
  Note that .tabset-dropdown was initially implemented using Glyphicons, which were dropped
  in BS4. The idea behind the "equivalent" BS4 CSS derives from how carets are implemented
  https://github.com/twbs/bootstrap/blob/099860/scss/mixins/_caret.scss#L1
-->
<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active-dropdown-tab:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active-dropdown-tab:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active-dropdown-tab {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Siobhon Egan</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="cv.html">
    <span class="fa fa-id-card"></span>
     
    CV
  </a>
</li>
<li>
  <a href="publications.html">
    <span class="fa fa-book"></span>
     
    Publications
  </a>
</li>
<li>
  <a href="tickID.html">
    <span class="fa fa-search"></span>
     
    Tick ID
  </a>
</li>
<li>
  <a href="contact.html">
    <span class="fa fa-at"></span>
     
    Contact
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CV</h1>

</div>


<p><span class="badge badge-primary">May 2020</span></p>
<p><strong>Research interests:</strong> Taxonomy and biology of ticks (Ixodida), Ecology, One Health, Parasitology, Wildlife Biology, Molecular Biology, Infectious Disease, Bioinformatics, Epidemiology.</p>
<div id="education" class="section level1">
<h1><i class="fas fa-graduation-cap"></i> Education</h1>
<p><strong>Doctor of Philosophy</strong> <em>Murdoch University</em> <strong>|</strong> 2018-2021</p>
<ul>
<li>Ecology of ticks and tick-borne diseases in Australia: A One Health approach.</li>
<li>Supervisors: Dr. Charlotte Oskam, Prof. Peter Irwin, Prof. Una Ryan, and Prof. Peter Banks.</li>
<li>Use of molecular tools to investigate the epidemiology of tick (and other vector) associated microbes (bacteria and protozoa) present in Australian wildlife.</li>
</ul>
<p><strong>Paid Internship</strong> <em>Pawsey Supercomputing Centre/CSIRO</em> <strong>|</strong> Summer 2017-2018</p>
<ul>
<li>Meta-analysis of Australian tick microbiomes</li>
<li>Supercomputing internship program sponsored by Pawsey and CSIRO</li>
<li>A video of my experiences as an intern available <a href="https://www.youtube.com/watch?v=4kf2OR9lEww&amp;t=1s">here</a></li>
</ul>
<p><strong>Bachelor of Science Honours</strong> <em>Murdoch University</em> <strong>|</strong> 2017</p>
<ul>
<li>Major in molecular biology First class</li>
<li>Profiling the microbiome of ticks parasitising Australian bandicoots <a href="http://researchrepository.murdoch.edu.au/id/eprint/40003">copy available here</a></li>
<li>Supervisors: Dr. Charlotte Oskam and Prof. Peter Irwin</li>
<li>Application of next generation sequencing to characterise the bacterial composition of tick (Acari: Ixodida) and tissues samples</li>
</ul>
<p><strong>Bachelor of Science</strong> <em>Murdoch University</em> <strong>|</strong> 2014 - 2016</p>
<ul>
<li>Double major in Animal Health and Conservation &amp; Wildlife Biology</li>
<li>Distinction average</li>
<li>Independent study contract in final semester – Small mammal camera trapping in Dryandra woodlands. Supervised by A/Prof Peter Spencer (Murdoch University) and Mark Cowan (Department of Parks and Wildlife WA).</li>
</ul>
</div>
<div id="grants-awards" class="section level1">
<h1><i class="fas fa-award"></i> Grants &amp; Awards</h1>
<p>Holsworth Wildlife Research Endowment awarded October 2020</p>
<p>Ticks and tick-borne pathogens (TTP10) conference student grant awarded February 2020 (postponed due to COVID-19)</p>
<p>Concepts in Parasitology course sponsorship (Bayer Australia) awarded November 2019</p>
<p>Holsworth Wildlife Research Endowment awarded July 2019</p>
<p>Australian Society for Parasitology Conference Travel Grant awarded July 2019</p>
<p>Australian Society for Parasitology Conference Travel Grant awarded September 2018</p>
<p>Holsworth Wildlife Research Endowment awarded June 2018</p>
<p>Research Training Program (RTP) Scholarship 2018-2021</p>
<p>Best poster design 2017 at the Combined Biological Sciences Meeting.</p>
<p>Myrtle AB Lamb Honours Scholarship awarded March 2017</p>
</div>
<div id="publications" class="section level1">
<h1><i class="far fa-file-alt"></i> Publications</h1>
<div id="articles" class="section level2">
<h2>Articles</h2>
<p><strong>Egan, S.</strong>, Ruiz-Aravena, M., Austen, J., Barton, X., Comte, S., Hamilton, D., Hamede, R., Ryan, U., Irwin, P., Jones, M., and Oskam, C. <em>Accepted</em>. Blood parasites in endangered wildlife: trypanosomes discovered during a survey of haemoprotozoa from the Tasmanian devil. <em>Pathogens</em>.</p>
<p>Boulet, C., <strong>Egan, S.</strong>, Jones, L., Jonsdottir, T., Power, M., Ralph, S., Tran, L., and White, M. 2020. Parasitravaganza 2020: Insights into a Virtual Parasitology Conference. <em>Trends in Parasitology</em>. DOI: <a href="https://doi.org/10.1016/j.pt.2020.09.009">10.1016/j.pt.2020.09.009</a></p>
<p>Austen, J., Van Kampen E., <strong>Egan, S.,</strong>, O’Dea, M., Jackson, B., Ryan, U., Irwin, P., and Prada, D. 2020. First report of <em>Trypanosoma dionisii</em> (Trypanosomatidae) identified in Australia. <em>Parasitology</em>. DOI <a href="https://doi.org/10.1017/S0031182020001845">10.1017/S0031182020001845</a></p>
<p><strong>Egan, S.</strong>, Taylor, C., Austen, J., Banks, P., Alhstrom, L., Ryan, U., Irwin, P., and Oskam C. 2020. Molecular identification of the <em>Trypanosoma</em> (<em>Herpetosoma</em>) <em>lewisi</em> clade in black rats (<em>Rattus rattus</em>) from Australia. <em>Parasitology Research</em>. DOI <a href="https://doi.org/10.1007/s00436-020-06653-z">10.1007/s00436-020-06653-z</a></p>
<p><strong>Egan, S.</strong>, Loh, S-M., Banks, P., Gillet, A., Alhstrom, L., Ryan, U., Irwin, P., and Oskam C. 2020. Bacterial community profiling highlights complex diversity and novel organisms in wildlife ticks. <em>Ticks and Tick-Borne Diseases</em> 11(3). DOI <a href="https://doi.org/10.1016/j.ttbdis.2020.101407">10.1016/j.ttbdis.2020.101407</a></p>
<p>Evans, M., <strong>Egan, S.</strong>, Irwin, P., and Oskam, C. 2019. Automatic Barcode Gap Discovery reveals large COI intraspecific divergence in Australian Ixodidae. <em>Zootaxa</em> 4652(2). DOI <a href="https://doi.org/10.11646/zootaxa.4656.2.13">10.11646/zootaxa.4656.2.13</a></p>
<p>Irwin, P., <strong>Egan, S.</strong>, Greay, T., and Oskam C. 2018. Bacterial tick-associated infections in Australia: Current studies and future directions. <em>Microbiology Australia</em> 39(4): 200-202. DOI <a href="https://doi.org/10.1071/MA18063">10.1071/MA18063</a></p>
<p>Loh, S-M., <strong>Egan, S.</strong>, Gillet, A., Banks, P., Ryan U., Irwin, P., and Oskam, C. 2018. Molecular surveillance of piroplasms in ticks from small and medium-sized urban and peri-urban mammals in Australia. <em>International Journal of Parasitology: Parasites and Wildlife</em> 7(2). DOI <a href="https://doi.org/10.1016/j.ijppaw.2018.05.005">10.1016/j.ijppaw.2018.05.005</a></p>
</div>
<div id="conference-presentations" class="section level2">
<h2>Conference presentations</h2>
<p><strong>Egan, S.</strong>, Fontaine, J., Ryan, U., Irwin. P, and Oskam C. (Postponed in 2020, rescheduled to August 2021). Molecular and ecological insights into the ornate kangaroo tick, <em>Amblyomma triguttatum</em>. <em>10th Tick and Tick-borne Pathogen Conference (TTP10)</em>, Romania.</p>
<p><strong>Egan, S.</strong>, Fontaine, J., Veber, W., Ruthof, K., Ryan, U., Irwin, P., and Oskam, C. 2019. The ecology of Amblyomma triguttatum triguttatum in Western Australia. <em>Australian Society for Parasitology Annual Conference</em>, Adelaide, South Australia.</p>
<p><strong>Egan, S.</strong>, Evans, M., Irwin, P., and Oskam, C. 2019. Molecular barcoding of Australian hard ticks (Ixodidae). <em>Australian Society for Parasitology Annual Conference</em>, Adelaide, South Australia. DOI <a href="https://doi.org/10.13140/RG.2.2.26208.02565">10.13140/RG.2.2.26208.02565</a></p>
<p><strong>Egan, S.</strong>, Siew-May, L., Gillet, A., Banks, P., Ryan U., Irwin, P., and Oskam, C. 2018. Ticks parasitizing Australian wildlife and surveillance of potential tick-borne pathogens. <em>Australian Society for Parasitology Annual Conference</em>, Melbourne, Victoria. DOI <a href="https://doi.org/10.13140/RG.2.2.29305.31841">10.13140/RG.2.2.29305.31841</a></p>
</div>
<div id="poster-presentations" class="section level2">
<h2>Poster presentations</h2>
<p><strong>Egan, S.</strong>, Loh, S-M., Banks, P., Gillett, A., Ahlstrom, L., Ryan, U., Irwin. P., and Oskam C. (Postponed in 2020, rescheduled to August 2021). The Australian tick microbiome: highlighting the diversity down under. <em>10th Tick and Tick-borne Pathogen Conference (TTP10)</em>, Romania.</p>
<p><strong>Egan, S.</strong>, Greay, T., Gofton, A., Ryan U., Irwin, P., and Oskam, C. 2018. What’s in a tick? Meta-analysis of microbial diversity in Australian ticks. <em>Global Microbiome II 2018 (West Coast Microbiome Network)</em>, Perth, Western Australia. DOI <a href="https://doi.org/10.13140/RG.2.2.34595.71200">10.13140/RG.2.2.34595.71200</a></p>
<p><strong>Egan, S.</strong>, Siew-May, L., Gofton, A., Greay, T., Banks, P., Irwin, P., and Oskam, C. 2017. Are bandicoot ticks a reservoir for novel bacteria? <em>Combined Biological Sciences Meeting</em>, Perth, Western Australia. DOI <a href="https://doi.org/10.13140/RG.2.2.34524.49285">10.13140/RG.2.2.34524.49285</a></p>
</div>
</div>
<div id="employment" class="section level1">
<h1><i class="fas fa-briefcase"></i> Employment</h1>
<ul>
<li>Present - 2017 | Demonstrator (casual)
<ul>
<li>Murdoch University, Perth, Australia</li>
<li>Units: Veterinary Parasitology (VET278), Veterinary Microbiology (VET211), Forensic DNA Analysis (BIO352)</li>
</ul></li>
<li>2018 | Research assistant (casual)
<ul>
<li>Murdoch University and Department of Primary Industries, Perth, Australia</li>
<li>Project supervisors: Dr Peter Adams and Dr. Charlotte Oskam</li>
<li>Dietary analysis of feral pigs using Next Generation Sequencing</li>
</ul></li>
</ul>
</div>
<div id="service-presentations-misc." class="section level1">
<h1><i class="fas fa-hands-helping"></i> Service, presentations &amp; misc.</h1>
<p>Conference Organising Committee <strong>|</strong> July 2020<br />
Australian Society for Parasitology<br />
Online Parasitravaganza workshop and conference</p>
<p>Tick Biology and Identification lecture and laboratory <strong>|</strong> 2019 &amp; 2020<br />
MICR5835 Vectors of Infectious Diseases and Vector Control (masters course)<br />
The University of Western Australia</p>
<p>Student Chapter Committee <strong>|</strong> 2018-2019<br />
Wildlife Disease Association Australiasia</p>
<p>Murdoch University Open Day <strong>|</strong> July 2019<br />
Parasitology display</p>
<p>STEM &amp; Supercomputers - career’s night <strong>|</strong> July 2018<br />
Pawsey Supercomputing Centre, Perth, Western Australia<br />
<a href="https://pawsey.org.au/stem-supercomputers-pawseys-first-careers-night/">Event write up</a></p>
<p>Perth Royal Show <strong>|</strong> September 2017<br />
Parasitology Life Undercover Display</p>
<p>Field &amp; Lab Assistant <strong>|</strong> December 2014-July 2016<br />
Murdoch University, Parasitology Department</p>
<p>Secretary <strong>|</strong> July 2014-July 2016<br />
Western Australia Wildlife Rehabilitation Council Inc.</p>
<p>Animal attendant <strong>|</strong> March 2014-July 2015<br />
Little Paws Rescue</p>
</div>
<div id="training-skills" class="section level1">
<h1><i class="fas fa-book-open"></i> Training &amp; Skills</h1>
<p><strong>Short courses</strong></p>
<p><code>A selection of relevant short courses and workshops</code></p>
<ul>
<li>Concepts in Parasitology <strong>|</strong> November-December 2019
<ul>
<li>Held at the Australian National University, Canberra &amp; Kioloa, Australia.</li>
<li>Coordinated by the Australian Society for Parasitology.</li>
</ul></li>
<li>ASMI BioInfoSummer <strong>|</strong> December 2018
<ul>
<li>Held at the University of Western Australia, Perth, Australia.</li>
<li>Coordinated by the Australian Mathematical Sciences Institute (AMSI).</li>
</ul></li>
<li>5th National Postgraduate Training Workshop in Systematics <strong>|</strong> July 2018
<ul>
<li>Held at The University of Adelaide, Adelaide, Australia.</li>
<li>Coordinated by the Australian Centre for Evolutionary Biology &amp; Biodiversity.</li>
</ul></li>
<li>Scientific use of animals <strong>|</strong> March 2018
<ul>
<li>Held at Murdoch University, Perth, Australia.</li>
<li>Coordinated by the Animal Ethics Committee, Murdoch University, Perth, Australia.</li>
</ul></li>
<li>Super computing and cloud resources <strong>|</strong> June 2017
<ul>
<li>Held at Murdoch University, Perth, Australia.</li>
<li>Coordinated by the Pawsey Supercomputing Centre.</li>
</ul></li>
</ul>
<p><strong>Certificates &amp; training courses</strong></p>
<ul>
<li>Safety in research training (Murdoch University)</li>
<li>Chemical safety training (ChemWatch)</li>
<li>Introduction to working with animals in research and teaching (University of Western Australia and Murdoch University)</li>
<li>Animal handling training (Department of Parks and Wildlife, WA)</li>
<li>Various wildlife workshops (facilitated by Department of Parks and Wildlife, WA and various Wildlife Centres)</li>
</ul>
<p><strong>Technical skills</strong></p>
<ul>
<li>RStudio - intermediate-advanced knowledge</li>
<li>Biology and Ecology modeling programs – PRIMER, SOCPROG</li>
<li>Molecular laboratory techniques – inc. Next Generation Sequencing, Sanger Sequencing, qPCR</li>
<li>Microscopy techniques</li>
<li>Computational Biology – Unix platform</li>
<li>Bioinformatics and phylogenetic analysis (programs include MEGA, Geneious, MEGAN)</li>
<li>Coding languages – Bash (intermediate-advanced), python (intermediate)</li>
<li>Wildlife camera trapping</li>
</ul>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
    $(this).siblings().removeClass("active-dropdown-tab");
    $(this).addClass("active-dropdown-tab");
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
