<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Siobhon Egan">
<meta name="dcterms.date" content="2021-02-13">
<meta name="description" content="Notes on models for phylogenetics">

<title>Siobhon Egan - Phylogenetic models</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
<meta name="twitter:title" content="Siobhon Egan - Phylogenetic models">
<meta name="twitter:description" content="Notes on models for phylogenetics">
<meta name="twitter:image" content="https://www.siobhon-egan.github.io/images/profile.png">
<meta name="twitter:creator" content="@siobhon_egan">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image-height" content="313">
<meta name="twitter:image-width" content="320">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Siobhon Egan</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../cv.html">
 <span class="menu-text">CV</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../publications.html">
 <span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../contact.html">
 <span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html">
 <span class="menu-text">Blog</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/siobhon-egan/siobhon-egan.github.io"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/siobhon_egan"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#rate-variation" id="toc-rate-variation" class="nav-link active" data-scroll-target="#rate-variation">Rate variation</a></li>
  <li><a href="#programs" id="toc-programs" class="nav-link" data-scroll-target="#programs">Programs</a></li>
  </ul>
</nav>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Phylogenetic models</h1>
  <div class="quarto-categories">
    <div class="quarto-category">genomics</div>
    <div class="quarto-category">phylogenetics</div>
  </div>
  </div>

<div>
  <div class="description">
    <p>Notes on models for phylogenetics</p>
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Siobhon Egan </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 13, 2021</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<p><span class="badge badge-info">Phylogenetics - overview of model types</span></p>
<p>An excellent summary of the different models (listed below) is directly sourced from Evolution and Geonomics Workshop (<a href="http://evomics.org/resources/substitution-models/nucleotide-substitution-models/">link here</a>).</p>
<p>Highly recommend you check out the rest of the website for some great resources and detailed information &amp; tutorials.</p>
<p>The use of maximum likelihood (ML) algorithms in developing phylogenetic hypotheses requires a model of evolution. The frequently used General Time Reversible (GTR) family of nested models encompasses 64 models with different combinations of parameters for DNA site substitution. The models are listed here from the least complex to the most parameter rich.</p>
<p>Some common programs to carry out phylogenetic analysis include:</p>
<ul>
<li><a href="http://nbisweden.github.io/MrBayes/">MrBayes</a></li>
<li><a href="https://paup.phylosolutions.com/">PAUP*</a></li>
<li><a href="http://www.beast2.org">BEAST v2</a></li>
<li><a href="http://www.atgc-montpellier.fr/phyml/">PhyML v3</a></li>
<li><a href="http://evomics.org/resources/software/molecular-evolution-software/paml/">PAML</a></li>
<li><a href="https://sco.h-its.org/exelixis/software.html">RAxML</a></li>
<li><a href="http://www.iqtree.org/">IQ-TREE</a></li>
</ul>
<p>The below also include the settings if the model is available in these programs. <a href="http://phylobotanist.blogspot.com/2016/06/implementation-of-substitution-models.html">Some good info here</a>.</p>
<p><strong>Jukes-Cantor (JC/JC69)</strong> equal base frequencies, all substitutions equally likely (Jukes and Cantor 1969)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" checked="">MrBayes <code>nst=1</code></li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>aaaaaa</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>aaaaaa</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="" checked="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Felsenstein 1981 (F81)</strong> variable base frequencies, all substitutions equally likely (Felsenstein 1981)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" checked="">MrBayes <code>nst=1</code></li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>aaaaaa</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>aaaaaa</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST (check)</li>
<li><input type="checkbox" disabled="" checked="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Kimura 2-parameter (K80/K2)</strong> equal base frequencies, one transition rate and one transversion rate (Kimura 1980)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" checked="">MrBayes <code>nst=2</code></li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>abaaba</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>abbbba</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="" checked="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Hasegawa-Kishino-Yano (HKY)</strong> variable base frequencies, one transition rate and one transversion rate (Hasegawa et. al.&nbsp;1985). Note this model is very similar to K80 however allows for variable base frequencies. It is also commonly the default model in many programs.</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" checked="">MrBayes <code>nst=2</code></li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>abaaba</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>abbbba</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="" checked="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Tamura-Nei (TrN/TN93)</strong> variable base frequencies, equal transversion rates, variable transition rates (Tamura Nei 1993)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="">MrBayes</li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>abaaea</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>abbbbf</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="" checked="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Kimura 3-parameter (K3P)</strong> variable base frequencies, equal transition rates, two transversion rates (Kimura 1981)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="">MrBayes</li>
<li><input type="checkbox" disabled="" checked="">PAUP (<code>abccba</code>)</li>
<li><input type="checkbox" disabled="" checked="">PAML (<code>abccba</code>)</li>
<li><input type="checkbox" disabled="">BEAST</li>
<li><input type="checkbox" disabled="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Transition model (TIM)</strong> variable base frequencies, variable transition rates, two transversion rates</p>
<ul class="task-list">
<li><input type="checkbox" disabled="">MrBayes</li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>abccea</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>abccbe</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="">RAxML (check)</li>
<li><input type="checkbox" disabled="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Transversion model (TVM)</strong> variable base frequencies, variable transversion rates, transition rates equal</p>
<ul class="task-list">
<li><input type="checkbox" disabled="">MrBayes</li>
<li><input type="checkbox" disabled="" checked="">PAUP (<code>abcdbe</code>)</li>
<li><input type="checkbox" disabled="" checked="">PAML (<code>abcdea</code>)</li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>Symmetrical model (SYM)</strong> equal base frequencies, symmetrical substitution matrix (A to T = T to A) (Zharkikh 1994)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" checked="">MrBayes <code>nst=6</code></li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>abcdef</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>abcdef</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<p><strong>general time reversible (GTR)</strong> variable base frequencies, symmetrical substitution matrix (e.g., Lanave et al.&nbsp;1984, Tavare 1986, Rodriguez et. al.&nbsp;1990)</p>
<ul class="task-list">
<li><input type="checkbox" disabled="" checked="">MrBayes <code>nst=6</code></li>
<li><input type="checkbox" disabled="" checked="">PAUP <code>abcdef</code></li>
<li><input type="checkbox" disabled="" checked="">PAML <code>abcdef</code></li>
<li><input type="checkbox" disabled="" checked="">BEAST</li>
<li><input type="checkbox" disabled="" checked="">PhyML</li>
<li><input type="checkbox" disabled="" checked="">IQ-TREE</li>
</ul>
<section id="rate-variation" class="level3">
<h3 class="anchored" data-anchor-id="rate-variation">Rate variation</h3>
<p>In addition to models describing the rates of change from one nucleotide to another, there are models to describe rate variation among sites in a sequence. The following are the two most commonly used models and are generally available across all platforms and programs.</p>
<p><strong>gamma distribution (G)</strong> gamma distributed rate variation among sites</p>
<p><strong>proportion of invariable sites (I)</strong>* extent of static, unchanging sites in a dataset</p>
<hr>
</section>
<section id="programs" class="level1">
<h1>Programs</h1>
<p>Some well resourced programs to carry out model selection include:</p>
<ul>
<li><a href="https://www.megasoftware.net/">MEGA</a></li>
<li><a href="http://www.atgc-montpellier.fr/phyml/">PhyML</a></li>
<li><a href="https://github.com/ddarriba/jmodeltest2">jmodeltest</a></li>
<li><a href="https://doi.org/10.1038/nmeth.4285">ModelFinder</a> which is implemented in <a href="http://www.iqtree.org/">IQ-TREE</a></li>
</ul>
<p>Depending on the program most will produce two sets of ‘scores’ to assess the models. the Akaike information criterion (AIC) and Bayesian information criterion (BIC). The lower the score the more support there is for a particular model. There are different reasons for choosing the best model based on the AIC or BIC scores. Generally the BIC score is used in most cases, however if in doubt its always best to do some more research to understand why you chose that value. Some references for more information on these criterion available here <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2925852/">Luo A. <em>et al.</em> 2010</a>; <a href="https://www.annualreviews.org/doi/pdf/10.1146/annurev.ecolsys.36.102003.152633">Sullivan J. and Joyce P. 2005</a>,<a href="https://besjournals.onlinelibrary.wiley.com/doi/full/10.1111/2041-210X.12541">Brewer M. J. <em>et al.</em> 2016</a>.</p>
<p>Example output from MEGA7 using Model Selection feature <img src="mega7-ml.png" class="img-fluid" alt="MEGA7_ModelSelection"></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://utteranc.es/client.js" repo="siobhon-egan/siobhon-egan.github.io" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Copyright 2022, Siobhon Egan</div>   
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/siobhon-egan/siobhon-egan.github.io">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/siobhon_egan">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>