<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Siobhon Egan">
<meta name="dcterms.date" content="2021-07-30">
<meta name="description" content="Notes for dealing with metabarcoding sequence data.">

<title>Siobhon Egan - Metabarcoding</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
<meta name="twitter:title" content="Siobhon Egan - Metabarcoding">
<meta name="twitter:description" content="Notes for dealing with metabarcoding sequence data.">
<meta name="twitter:image" content="https://www.siobhon-egan.github.io/images/profile.png">
<meta name="twitter:creator" content="@siobhon_egan">
<meta name="twitter:card" content="summary_large_image">
<meta name="twitter:image-height" content="313">
<meta name="twitter:image-width" content="320">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Siobhon Egan</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../about.html">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../cv.html">
 <span class="menu-text">CV</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../publications.html">
 <span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../contact.html">
 <span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html">
 <span class="menu-text">Blog</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/siobhon-egan/siobhon-egan.github.io"><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/siobhon_egan"><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#metabarcoding" id="toc-metabarcoding" class="nav-link active" data-scroll-target="#metabarcoding">Metabarcoding</a>
  <ul class="collapse">
  <li><a href="#additional-analysis" id="toc-additional-analysis" class="nav-link" data-scroll-target="#additional-analysis">Additional analysis</a></li>
  <li><a href="#rshiny" id="toc-rshiny" class="nav-link" data-scroll-target="#rshiny">RShiny</a></li>
  <li><a href="#sample-size-calculator" id="toc-sample-size-calculator" class="nav-link" data-scroll-target="#sample-size-calculator">Sample size calculator</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Metabarcoding</h1>
  <div class="quarto-categories">
    <div class="quarto-category">genomics</div>
    <div class="quarto-category">metabarcoding</div>
  </div>
  </div>

<div>
  <div class="description">
    <p>Notes for dealing with metabarcoding sequence data.</p>
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Siobhon Egan </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 30, 2021</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="metabarcoding" class="level1">
<h1>Metabarcoding</h1>
<p><span class="badge badge-info">Analysis of amplicon metabarcoding next generation sequencing data</span></p>
<p>ðŸ”— <strong>Useful resources</strong></p>
<ul>
<li><a href="https://github.com/mdozmorov/Microbiome_notes">Microbiome notes by Mikhail Dozmorov</a></li>
<li><a href="https://rdrr.io/category/biocview/Metagenomics/">Metagenomics R packages</a></li>
<li><a href="https://microbiome.github.io/tutorials/">Introduction to the microbiome R package by Leo Lahti, Sudarshan Shetty et al.</a>, <a href="https://microbiome.github.io/tutorials/Core.html">core microbiome</a> - Tools for microbiome analysis; with multiple example data sets from published studies; extending the phyloseq class. The package is in Bioconductor and aims to provide a comprehensive collection of tools and tutorials, with a particular focus on amplicon sequencing data.</li>
<li><a href="https://towardsdatascience.com/how-to-give-life-to-your-microbiome-data-using-plotly-r-1892281183cf">How to give life to your microbiome data by Ruth Schmidt</a> - A tutorial on how to use Plotlyâ€™s R graphing library for microbiome data analysis and visualization. Also see <a href="https://github.com/ruthlys/">GitHub</a> for more useful info.</li>
<li><a href="https://github.com/dgiguer/omicplotR">omicplotR</a> - An R package to visualize high-throughput sequencing data.</li>
<li><a href="https://micca.readthedocs.io/en/1.7.2/index.html">MICrobial Community Analysis (micca)</a> - software pipeline for the processing of amplicon sequencing data, from raw sequences to OTU tables, taxonomy classification and phylogenetic tree inference. The pipeline can be applied to a range of highly conserved genes/spacers, such as 16S rRNA gene, Internal Transcribed Spacer (ITS) 18S and 28S rRNA.</li>
<li><a href="https://vaulot.github.io/tutorials/Phyloseq_tutorial.html#content">phyloseq tutoial by Daniel Vaulot</a> - This document explains the use of the phyloseq R library to analyze metabarcoding data. -<a href="https://github.com/otagoedna">Otago Uni eDNA GitHub page</a> -<a href="https://rachaellappan.github.io/VL-QIIME2-analysis/">Amplicon analysis with QIIME2 - VL microbiome project by Rachael Lappan</a></li>
<li><a href="https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html#abstract">Workflow for Microbiome Data Analysis: from raw reads to community analyses by Benjamin J Callahan</a></li>
<li><a href="https://microbiomer.vallenderlab.org/">MicrobiomeR</a> - An R package for microbiome analysis that incorporates phyloseq, metacoder, taxa, and microbiome in order to standardize and simplify common microbiome workflows.</li>
<li><a href="https://chiliubio.github.io/microeco/">microeco: An R package for data mining in microbial community ecology by Chi Liu, Minjie Yao</a></li>
<li><a href="https://madsalbertsen.github.io/ampvis2/index.html">ampvis2</a> - ampvis2 is an R-package to conveniently visualise and analyse 16S rRNA amplicon data in different ways.</li>
<li><a href="https://github.com/umerijaz/microbiomeSeq">microbiomeseq</a> - This R package is developed to enhance the available statistical analysis procedures in R by providing more analysis produre and visualisation of results for microbial communities data obtained from 16S rRNA.</li>
<li><a href="https://grunwaldlab.github.io/analysis_of_microbiome_community_data_in_r/index.html">Analysis of Microbiome Community Data in R by Grunwald lab</a></li>
<li><a href="https://github.com/jpearl01/mcsmrt">MCSMRT Microbiome Classifier for SMRT PacBio data</a> - a tool to cluster PacBio FL16S amplicon microbiome sequences into Operational Taxonomic Units (OTU) and assign species level taxonomic classifications. Outputs include a table of read counts assigned to each OTU centroid sequence (per sample) with corresponding taxonomic lineage, and a a table of read specific metrics (e.g., CCS count, expected error, length, primer matching result, etc.).</li>
<li><a href="https://github.com/BioplatformsAustralia/bpaotu">Bioplatforms Australia - Operational taxonomic unit (OTU) query system</a> - BPA-OTU is a web-based portal into Operational Taxonomic Unit (OTU) data, developed to access data from the Australian Microbiome.</li>
<li><a href="https://github.com/martinostrowski/metagenome">AusMicrobiome metagenome</a> - <em>In development</em> An assembly and annotation pipeline using standardised tools to characterise shotgun metagenomes produced by the Marine Microbes and Australian Microbiome Project.</li>
</ul>
<section id="additional-analysis" class="level2">
<h2 class="anchored" data-anchor-id="additional-analysis">Additional analysis</h2>
<ul>
<li><a href="https://benjjneb.github.io/decontam/vignettes/decontam_intro.html">decontam R package</a> - R package that outlines benchmarking, and demonstrates the benefits of <code>decontam</code>-inating your data for more accurate profiling of microbial communities.</li>
<li><a href="https://app.pr2-primers.org/">The PR3 primer database</a> - A database of eukaryotic rRNA primers and primer sets for metabarcoding studies compiled from the literature.</li>
<li><a href="https://evayiwenwang.github.io/Managing_batch_effects/">Managing Batch Effects in Microbiome Data by Yiwen Wang, Kim-Anh LÃª Cao</a></li>
</ul>
</section>
<section id="rshiny" class="level2">
<h2 class="anchored" data-anchor-id="rshiny">RShiny</h2>
<ul>
<li><a href="https://github.com/Biodata-PT/Biome-Shiny">Biome-Shiny</a> - A Shiny R app for microbiome visualization.</li>
<li><a href="https://acnc-shinyapps.shinyapps.io/DAME/">Dynamic Assessment of Microbial Ecology (DAME)</a> - This R Shiny app is an open source platform that uses the R environment to perform microbial ecology data analyses (designed for QIIME1 output).</li>
<li><a href="https://compbiomed.github.io/animalcules-docs/index.html">animalcules</a> - animalcules is an R package/R Shiny app for utilizing up-to-date data analytics, visualization methods, and machine learning models to provide users an easy-to-use interactive microbiome analysis framework.</li>
<li><a href="https://github.com/zoecastillo/microbiomeExplorer">An R package for the analysis and visualization of microbial communities by Janina Reeder</a> - The microbiomeExplorer package and the Shiny application contained with it provides methods and visualizations to explore the results of 16S rRNA amplicon sequencing experiment.</li>
</ul>
</section>
<section id="sample-size-calculator" class="level2">
<h2 class="anchored" data-anchor-id="sample-size-calculator">Sample size calculator</h2>
<p>Links to some useful resources for calculating power and sample size in microbiome studies. Good for grant and project proposals to show you have thought about sample size etc.</p>
<p>Some general points to consider</p>
<ul>
<li>What population should I target?</li>
<li>How should I collect and store the samples? (CONTROLS!!!)</li>
<li>What am I targeting and how should i extract?</li>
<li>What sequencing approach should I use, amplicon or whole shotgun metagenome? Will there be PCR involved or do I want a pre-PCR library preparation method?</li>
<li>How deeply should I sequence?</li>
</ul>
<p><strong>References</strong></p>
<ul>
<li>Mattiello F, Verbist B, Faust K, Raes J, Shannon WD, Bijnens L, Thas O. A web application for sample size and power calculation in case-control microbiome studies. Bioinformatics. 2016 Jul 1;32(13):2038-40. doi: <a href="https://doi.org/10.1093/bioinformatics/btw099">10.1093/bioinformatics/btw099</a></li>
<li>Kelly BJ, Gross R, Bittinger K, Sherrill-Mix S, Lewis JD, Collman RG, Bushman FD, Li H. Power and sample-size estimation for microbiome studies using pairwise distances and PERMANOVA. Bioinformatics. 2015 Aug 1;31(15):2461-8. doi: <a href="https://doi.org/10.1093/bioinformatics/btv183">10.1093/bioinformatics/btv183</a></li>
<li>Casals-Pascual C, GonzÃ¡lez A, VÃ¡zquez-Baeza Y, Song SJ, Jiang L, Knight R. Microbial Diversity in Clinical Microbiome Studies: Sample Size and Statistical Power Considerations. Gastroenterology. 2020 May;158(6):1524-1528. doi: <a href="https://doi.org/10.1053/j.gastro.2019.11.305">10.1053/j.gastro.2019.11.305</a>.</li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://utteranc.es/client.js" repo="siobhon-egan/siobhon-egan.github.io" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Copyright 2022, Siobhon Egan</div>   
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/siobhon-egan/siobhon-egan.github.io">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/siobhon_egan">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>